lib/fastlane/plugin/appcenter/actions/appcenter_fetch_devices_action.rb

```diff
@@ -0,0 +1,138 @@
+require 'fastlane/action'
+require 'csv'
+require 'faraday'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      APPCENTER_API_TOKEN = :APPCENTER_API_TOKEN
+      APPCENTER_OWNER_NAME = :APPCENTER_OWNER_NAME
+      APPCENTER_APP_NAME = :APPCENTER_APP_NAME
+      APPCENTER_DISTRIBUTE_DESTINATIONS = :APPCENTER_DISTRIBUTE_DESTINATIONS
+    end
+
+    class AppcenterFetchDevicesAction < Action
+      def self.run(params)
+        api_token = params[:api_token]
+        owner_name = params[:owner_name]
+        app_name = params[:app_name]
+        destinations = params[:destinations]
+
+        Actions.lane_context[SharedValues::APPCENTER_API_TOKEN] = api_token
+        Actions.lane_context[SharedValues::APPCENTER_OWNER_NAME] = owner_name
+        Actions.lane_context[SharedValues::APPCENTER_APP_NAME] = app_name
+        Actions.lane_context[SharedValues::APPCENTER_DISTRIBUTE_DESTINATIONS] = destinations
+
+        group_names = []
+        if destinations == '*'
+          UI.message("Looking up all distribution groups for #{owner_name}/#{app_name}")
+          distribution_groups = Helper::AppcenterHelper.fetch_distribution_groups(
+            api_token: api_token,
+            owner_name: owner_name,
+            app_name: app_name
+          )
+          UI.abort_with_message!("Failed to list distribution groups for #{owner_name}/#{app_name}") unless distribution_groups
+          distribution_groups.each do |group|
+            group_names << group['name']
+          end
+        else
+          group_names += destinations.split(',')
```

Comment:
Might want to trim each group here as well (unless you handled that elsewhere and I missed it). Some people will probably add spaces after their commas out of habit.

Reply:
Good point!